### PR TITLE
Add treasury to data source

### PR DESCRIPTION
**Pull request by Benzbeeb**
Fixed: #XXXX

## Please ensure the following requirements are met before submitting a pull request:

- [ ] The pull request is targeted against the correct target branch
- [ ] The pull request is linked to an issue with appropriate discussion and an accepted design OR is linked to a spec that describes the work.
- [ ] The pull request includes a description of the implementation/work done in detail.
- [ ] The pull request includes any and all appropriate unit/integration tests
- [ ] You have added a relevant changelog entry to `CHANGELOG_UNRELEASED.md`
- [ ] You have re-reviewed the files affected by the pull request (e.g. using the `Files changed` tab in the Github PR explorer)

### CODE REVIEW

File: x/oracle/client/cli/tx.go [hunk 1]
```diff
@@ -186,12 +197,21 @@ $ %s tx oracle create-data-source --name coingecko-price --description "The scri
...
+			coinStr, err := cmd.Flags().GetString(flagFeeLimit)
+			if err != nil {
+				return err
+			}
+
+			feeLimit, err := sdk.ParseCoinsNormalized(coinStr)
+			if err != nil {
+				return err
+			}
+
```
Comment (taobun) [inline x/oracle/client/cli/tx.go:209]: Add treasury flag too

**Review by taobun — CHANGES_REQUESTED** [review]

File: x/oracle/client/cli/tx.go [hunk 2]
```diff
@@ -270,14 +302,32 @@ $ %s tx oracle edit-data-source 1 --name coingecko-price --description The scrip
...
+			treasuryStr, err := cmd.Flags().GetString(flagTreasury)
+			if err != nil {
+				return err
+			}
+			var treasury sdk.AccAddress
+			treasury, err = sdk.AccAddressFromBech32(treasuryStr)
```
Comment (taobun) [inline x/oracle/client/cli/tx.go:319]: treasury, err := sdk.AccAddressFromBech32(treasuryStr) ?

File: x/oracle/client/cli/tx.go [hunk 3]
```diff
@@ -270,14 +302,32 @@ $ %s tx oracle edit-data-source 1 --name coingecko-price --description The scrip
 				return err
 			}
 
-			// TODO: Add tresury and fee flag
+			coinStr, err := cmd.Flags().GetString(flagFeeLimit)
```
Comment (taobun) [inline x/oracle/client/cli/tx.go:305]: Add `TODO: Support do-not-modify fee`

**Review by taobun — CHANGES_REQUESTED** [review]

File: flusher/flusher/db.py [hunk 4]
```diff
@@ -257,12 +262,7 @@ def Column(*args, **kwargs):
     "validator_votes",
     metadata,
     Column("block_height", sa.Integer, sa.ForeignKey("blocks.height"), primary_key=True),
-    Column(
-        "consensus_address",
-        sa.String,
-        sa.ForeignKey("validators.consensus_address"),
-        primary_key=True,
-    ),
+    Column("consensus_address", sa.String, sa.ForeignKey("validators.consensus_address"), primary_key=True,),
```
Comment (thebevrishot) [inline flusher/flusher/db.py:265]: Please remove unwanted `,`

File: flusher/flusher/init.py [hunk 5]
```diff
@@ -13,103 +13,157 @@
...
     engine.execute(
         tracking.insert(),
-        {
-            "chain_id": chain_id,
-            "topic": topic,
-            "replay_topic": replay_topic,
-            "kafka_offset": -1,
-            "replay_offset": -1,
-        },
+        {"chain_id": chain_id, "topic": topic, "replay_topic": replay_topic, "kafka_offset": -1, "replay_offset": -1,},
```
Comment (thebevrishot) [inline flusher/flusher/init.py:24]: please remove unwanted `,`

File: flusher/flusher/init.py [hunk 6]
```diff
@@ -13,103 +13,157 @@
...
+CREATE VIEW oracle_script_statistic_last_1_month
+AS
+  SELECT Avg(resolve_time - request_time) AS response_time,
+         Count(*)                         AS count,
+         oracle_scripts.id,
+         resolve_status
+  FROM   oracle_scripts
+         join requests
+           ON oracle_scripts.id = requests.oracle_script_id
+  WHERE  To_timestamp(requests.request_time) >= Now() - '1 month' :: interval
```
Comment (thebevrishot) [inline flusher/flusher/init.py:163]: should be `to_timestamp` or `TO_TIMESTAMP`

**Review by thebevrishot — APPROVED** [review]

File: x/oracle/client/cli/tx.go [hunk 7]
```diff
@@ -208,6 +238,8 @@ $ %s tx oracle create-data-source --name coingecko-price --description "The scri
 	cmd.Flags().String(flagName, "", "Name of this data source")
 	cmd.Flags().String(flagDescription, "", "Description of this data source")
 	cmd.Flags().String(flagScript, "", "Path to this data source script")
+	cmd.Flags().String(flagFeeLimit, "", "the maximum tokens that will be paid to all data source providers")
```
Comment (taobun) [inline x/oracle/client/cli/tx.go:241]: It's not a flagFeeLimit. It should be flagFee

File: x/oracle/client/cli/tx.go [hunk 8]
```diff
@@ -293,6 +343,8 @@ $ %s tx oracle edit-data-source 1 --name coingecko-price --description The scrip
 	cmd.Flags().String(flagName, types.DoNotModify, "Name of this data source")
 	cmd.Flags().String(flagDescription, types.DoNotModify, "Description of this data source")
 	cmd.Flags().String(flagScript, types.DoNotModify, "Path to this data source script")
+	cmd.Flags().String(flagFeeLimit, "", "the maximum tokens that will be paid to all data source providers")
```
Comment (taobun) [inline x/oracle/client/cli/tx.go:346]: ditto

**Review by taobun — CHANGES_REQUESTED** [review]

File: cmd/bandd/cmd/gends.go [hunk 9]
```diff
@@ -50,10 +50,17 @@ func AddGenesisDataSourceCmd(defaultNodeHome string) *cobra.Command {
 			if err != nil {
 				return fmt.Errorf("failed to unmarshal genesis state: %w", err)
 			}
+			feeLimit, err := sdk.ParseCoinsNormalized(args[4])
```
Comment (taobun) [inline cmd/bandd/cmd/gends.go:53]: fee

**Review by taobun — CHANGES_REQUESTED** [review]

**Review by thebevrishot — APPROVED** [review]

**Review by taobun — APPROVED** [review]

**Pull request merged**